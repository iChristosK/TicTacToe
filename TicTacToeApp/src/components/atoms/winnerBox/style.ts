import { StyleSheet } from "react-native";

export default StyleSheet.create({
    container: {
        flex: 1,
        alignItems: 'center',
        justifyContent: 'center',
        height: 60
    },
    winnerText: {
        fontSize: 34,
        fontWeight: 'bold',
        color: '#FC4D3C',
    },
});