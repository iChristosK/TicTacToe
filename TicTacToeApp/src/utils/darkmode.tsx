import React from 'react';
import {
  SafeAreaView,
  ScrollView,
  StatusBar,
  useColorScheme,
  View,
} from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';

const DarkMode = ({ children }: any) => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
    height: '100%',
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        <View
          style={{
            backgroundColor: isDarkMode ? Colors.black : Colors.white,
          }}>
          {children}
        </View>
      </ScrollView>
    </SafeAreaView>
  );
};

export default DarkMode;
