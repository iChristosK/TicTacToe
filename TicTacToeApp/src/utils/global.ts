export const LogoUrl =
  'https://camo.githubusercontent.com/56026388f0309ddf46bdbf5409d13e093295491960b0a8c8916168ecc5861972/68747470733a2f2f7261772e6769746875622e636f6d2f4568616247616d616c2f546963546163546f652f6d61737465722f436c69656e742f7372632f7265736f75726365732f696d616765732f7469632e676966';
export const checkWinnerString = 'Check Winner';
export const resetString = 'Reset';
export const tieString = 'TIE';
