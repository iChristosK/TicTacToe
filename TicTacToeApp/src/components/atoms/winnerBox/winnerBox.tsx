import { View } from 'react-native';
import CustomText from '../texts/CustomText';
import React from 'react';

import { style } from './style';

export interface WinnerProps {
  winner: string | null;
}

export const WinnerBox = ({ winner }: WinnerProps) => {
  const absoluteWin = winner === 'TIE' ? winner : winner;
  return (
    <View style={style.container}>
      {winner ? (
        <CustomText style={style.winnerText}>{absoluteWin} </CustomText>
      ) : null}
    </View>
  );
};
