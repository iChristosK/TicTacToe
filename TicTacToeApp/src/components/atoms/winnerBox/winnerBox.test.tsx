import React from 'react';
import renderer from 'react-test-renderer';
import { tieString } from '../../../utils/global';
import { WinnerBox } from './winnerBox';

describe('WinnerBox', () => {
    it('should render correctly when there is a winner', () => {
        const tree = renderer.create(<WinnerBox winner="X" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when it is a tie', () => {
        const tree = renderer.create(<WinnerBox winner={tieString} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render correctly when there is no winner', () => {
        const tree = renderer.create(<WinnerBox winner={null} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
