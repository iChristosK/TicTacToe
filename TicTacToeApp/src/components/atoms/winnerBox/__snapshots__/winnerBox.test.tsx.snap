// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WinnerBox should render correctly when it is a tie 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 34,
        "fontWeight": "bold",
      }
    }
  >
     
     IT IS A 
    TIE
     ! 
  </Text>
</View>
`;

exports[`WinnerBox should render correctly when there is a winner 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 60,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 34,
        "fontWeight": "bold",
      }
    }
  >
     
    X
     WON !
  </Text>
</View>
`;

exports[`WinnerBox should render correctly when there is no winner 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "height": 60,
      "justifyContent": "center",
    }
  }
/>
`;
